main.py

```diff
@@ -15,99 +15,105 @@
+        # If using a real servo, use the real servo object, otherwise use a mock servo object
         servo = (
             Servo(SERVO_PIN)
             if args.real_servo
             else Servo(SERVO_PIN, pin_factory=MockFactory(pin_class=MockPWMPin))
         )
         logger = MockLogger(LOGS_PATH, delete_log_file=not args.keep_log_file)
     else:
-        # If we are not running a simulation, then we will use the real hardware objects
+        # Use real hardware components
         servo = Servo(SERVO_PIN)
-        imu = IMU(PORT)
+        imu = IMU(IMU_PORT)
         logger = Logger(LOGS_PATH)
 
-    # Our data processing and apogee prediction stay the same regardless of whether we are running
-    # a simulation or not
-    data_processor = IMUDataProcessor()
-    apogee_predictor = ApogeePredictor()
+    return servo, imu, logger
 
-    # The context that will manage the airbrakes state machine
-    airbrakes = AirbrakesContext(servo, imu, logger, data_processor, apogee_predictor)
-
-    flight_display = FlightDisplay(airbrakes, sim_time_start, args)
 
+def run_flight_loop(
+    airbrakes: AirbrakesContext, flight_display: FlightDisplay, is_mock: bool
+) -> None:
+    """
+    Main flight control loop that runs until shutdown is requested or interrupted.
+    :param airbrakes: The airbrakes context managing the state machine.
+    :param flight_display: Display interface for flight data.
+    :param is_mock: Whether running in simulation mode.
+    """
     try:
-        airbrakes.start()  # Start the IMU and logger processes
-
-        # Setup our flight display
-        # This is what prints the flight data to the console in real time, we only do
-        # it when running the sim because printing a lot of things can significantly slow down
-        # the program
+        # Starts the airbrakes system and display
+        airbrakes.start()
         flight_display.start()
 
-        # This is the main loop that will run until we press Ctrl+C
         while not airbrakes.shutdown_requested:
-            # Update the airbrakes finite state machine
+            # Update the state machine
             airbrakes.update()
 
-            # Stop the sim when the data is exhausted:
-            if args.mock and not airbrakes.imu._data_fetch_process.is_alive():
+            # Stop the simulation when the data is exhausted
+            if is_mock and not airbrakes.imu.is_running:
                 break
+
+    # Handle user interrupt gracefully
     except KeyboardInterrupt:
-        if args.mock:
+        if is_mock:
             flight_display.end_sim_interrupted.set()
-    else:  # This is run if we have landed and the program is not interrupted (see state.py)
-        if args.mock:
-            # Usually the mock sim will stop itself due to data exhaustion, so we will actually
-            # hit the condition above, but if the data isn't exhausted, we will stop it here.
+    else:
```

Comment:
the comment which specified when the else block was executed is important context imo

Reply:
yeah its a good comment